morphic.js

```diff
@@ -7818,10 +7820,47 @@ MenuMorph.prototype.select = function (aMenuItem) {
     this.selection = aMenuItem;
 };
 
+MenuMorph.prototype.mouseLeave = function () {
+    if (this.parent instanceof MenuMorph) {
+        // this is a submenu so we vanish when lose focus
+        var mouses = this.world.hand.allMorphsAtPointer();
+        if (contains(mouses, this.parent)) {
+            var items = mouses.filter(function (morph)
+                                      {return morph instanceof MenuItemMorph});
+            if (items[0].doubleClickAction != this) {
+                this.destroy();
+            }
+        } else {
+            this.destroy();
+        }
+    }
+}
+            
+
 MenuMorph.prototype.destroy = function () {
+var myself = this;
+
     if (this.hasFocus) {
         this.world.keyboardReceiver = null;
     }
+
+    if ((this.parent instanceof MenuMorph) && this.world) {
+        // If I'm a submenu, un-highlight the item pointing to me.
+        var mouses = this.world.hand.allMorphsAtPointer(),
+        items = this.parent.children.filter(function (child) {
+            return (child instanceof MenuItemMorph) &&
+            (child.doubleClickAction instanceof MenuMorph) &&
+            !(contains(mouses, child)); });
```

Comment:
If the opening `{` is at the end of a line, then the closing `}` should be at the start of it's own line.